### PR TITLE
Using async routes

### LINKED ISSUE #428: Cosmetic change in 'swarm delete' output
```Deleting service helloworld ...```

should become

```Deleting service 'helloworld'…```

That's single quotes around the service name. And note that the three dots at the end are one character, not three. And there is no space before the dots.

Ping @ewoutp

### PR DESCRIPTION
**Pull request by ewoutp**
This change uses the task API and asynchronous route to start & stop services & components.

Dependencies:

- [x] giantswarm/client-go#76
- [x] giantswarm/client-go#77
- [x] giantswarm/client-go#79

Also fixes #428

### CODE REVIEW

File: start.go [hunk 1]
```diff
@@ -149,19 +152,20 @@ func startComponent(service, component string, detach, recursive bool) {
...
+	}
 
+	status, err := client.ComponentStatus(service, component, recursive)
+	if err != nil {
+		Quitf(defaultErrorMsgTemplate, maskAny(err))
+	}
+	finalStatus, err := status.Status()
+	if err != nil {
+		Quitf(defaultErrorMsgTemplate, maskAny(err))
+	}
```
Comment (zyndiecate) [inline start.go]: This is going to be nice because we can throw away a lot of status loop stuff. :)

**Comment by zyndiecate** [comment]
LGTM

**Comment by ewoutp** [comment]
With async routes the `--debug` output becomes really quiet.

```
$ swarm --debug stop
DEBUG: request ID 'zhsua00uwz9q8os3'
Stopping service busywait...
DEBUG: POST http://192.168.33.101:8080/v2/org/tester/env/dev/service/busywait/stop ()
DEBUG: << 200 OK
\ DEBUG: GET http://192.168.33.101:8080/v2/org/tester/env/dev/task/b4azi6z34fsvvd5a/waitfor
DEBUG: << 200 OK
DEBUG: GET http://192.168.33.101:8080/v1/org/tester/env/dev/service/busywait/status
DEBUG: << 200 OK
Service busywait is failed
```

File: stop.go [hunk 2]
```diff
@@ -86,8 +87,13 @@ func processStop(argument string, detached, recursive bool) {
 func stopService(service string, detached bool) {
 	Infof("Stopping service %s...\n", service)
 
-	if err := client.AppStop(service); err != nil {
-		Quitf(defaultErrorMsgTemplate, err)
+	task, err := client.ServiceStopAsync(service)
+	if err != nil {
+		if clientgo.IsNotFound(err) {
```
Comment (zyndiecate) [inline stop.go]: We should start using a helper method that knows all kind of errors and know how to process them. For each error we then just need to call this wise helper. #idea

File: utils.go [hunk 3]
```diff
@@ -187,6 +188,53 @@ func waitForStatus(getStatus func() userconfig.Status, isEndStatus func(userconf
...
+			Infof("  ") // Go ahead two spaces because these are remove in every spinner even
+			spinnerIndex := 0
+			for {
+				select {
+				case <-spinnerStopChan:
+					Infof("%s%s", backSpace, backSpace)
+					return
+				default:
+					// Wait a while
+					Infof("%s%s%s ", backSpace, backSpace, spinnerChars[spinnerIndex:spinnerIndex+1])
```
Comment (zyndiecate) [inline utils.go]: This code is just copied. Will we throw away the old code soon? Can we module that better somehow?

Comment (ewoutp) [inline utils.go]: Correct it is partially copied, the old methods will go away as soon as every wait is task based.

File: stop.go (hunk 2 above)
Comment (ewoutp) [inline stop.go]: +1, we just have to make sure that the feedback to the user remains unchanged. For all these errors, the feedback is (slightly) different. Some of that is on purpose, some of it has just happened.

**Comment by zyndiecate** [comment]
Still LGTM

**Comment by ewoutp** [comment]
Note: This should not be merged before the routes are available on all clusters.

File: int-tests/027-exec-usage.t [hunk 4]
```diff
@@ -10,8 +10,8 @@ Exec usage:
     https://docs.giantswarm.io/reference/cli/exec/
   
   Examples:
-    <instance-id> [command ...]    Execute a command in an instance
-    <instance-id> -d [command ...] Execute a command in detached mode (in the background)
+    <instance-id> [command \xe2\x80\xa6]    Execute a command in an instance (esc)
+    <instance-id> -d [command \xe2\x80\xa6] Execute a command in detached mode (in the background) (esc)
```
Comment (zyndiecate) [inline int-tests/027-exec-usage.t]: What is this sequence?

File: utils.go [hunk 5]
```diff
@@ -23,6 +24,7 @@ const (
 	minSameStatusCountLong   = 30 // wait up to 15sec (30 loops of 500ms) while getting the same status code (in case of starting a unit that is already in failed state)
 	spinnerChars             = "-\\|/-\\|/"
 	backSpace                = "\b"
+	dotDotDot                = "…"
```
Comment (zyndiecate) [inline utils.go]: Is this a special character? Are all terminals able to print that?

**Comment by zyndiecate** [comment]
Minor comments. Still LGTM

File: int-tests/027-exec-usage.t (hunk 4 above)
Comment (ewoutp) [inline int-tests/027-exec-usage.t]: Its the `...` character

File: utils.go (hunk 5 above)
Comment (ewoutp) [inline utils.go]: @marians Sure about this?

Comment (marians) [inline utils.go]: Non-unicode/utf8 terminals are probably not able to print it.

**Pull request merged**